File: CS/GridMVVMConditionalFormattingRules/FormatConditionSelector.cs
using System.Windows;
using System.Windows.Controls;

namespace GridMVVMConditionalFormattingRules {
    public class FormatConditionSelector : DataTemplateSelector {

        public DataTemplate BackgroundTemplate { get; set; }
        public DataTemplate FontTemplate { get; set; }
        public DataTemplate IconTemplate { get; set; }

        public override DataTemplate SelectTemplate(object item, DependencyObject container) {
            FormattingRule rule = item as FormattingRule;
            if(rule == null) return null;
            switch(rule.Type) {
                case FormattingType.Font:
                    return FontTemplate;
                case FormattingType.Background:
                    return BackgroundTemplate;
                case FormattingType.Icon:
                    return IconTemplate;
            }
            return null;
        }
    }
}

File: VB/GridMVVMConditionalFormattingRules/FormatConditionSelector.vb
Imports System.Windows
Imports System.Windows.Controls

Namespace GridMVVMConditionalFormattingRules
	Public Class FormatConditionSelector
		Inherits DataTemplateSelector

		Public Property BackgroundTemplate() As DataTemplate
		Public Property FontTemplate() As DataTemplate
		Public Property IconTemplate() As DataTemplate

		Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
			Dim rule As FormattingRule = TryCast(item, FormattingRule)
			If rule Is Nothing Then
				Return Nothing
			End If
			Select Case rule.Type
				Case FormattingType.Font
					Return FontTemplate
				Case FormattingType.Background
					Return BackgroundTemplate
				Case FormattingType.Icon
					Return IconTemplate
			End Select
			Return Nothing
		End Function
	End Class
End Namespace